sql/ballots/ballots.go

```diff
@@ -0,0 +1,118 @@
+package ballots
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+
+	"github.com/spacemeshos/go-spacemesh/codec"
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/sql"
+)
+
+func decodeBallot(id types.BallotID, sig, pubkey, body *bytes.Reader) (*types.Ballot, error) {
+	sigBytes := make([]byte, sig.Len())
+	if _, err := sig.Read(sigBytes); err != nil {
+		if err != io.EOF {
+			return nil, fmt.Errorf("copy sig %w", err)
+		}
+		sigBytes = nil
```

Comment:
i made it this way because for testing i don't need signature and pubkey, can change it to a stricter validation

Reply:
actually, it is very inconvenient to use valid signature and pubkey for testing, so i would rather leave it as it is, and enforce validation in other places (like it works now)

Reply:
hmmm... for me, the biggest benefit of using sql-like database is the guarantee on data integrity. if we cannot trust that whatever in the database is correct, then it lost a lot of its value to me.

we should have NOT NULL constraints for these two fields for ballot.

Reply:
it is a benefit for sure but not the most significant. i think that it doesn't matter much because we didn't enforce it with leveldb, so for me it is irrelevant if it will be enforced with sqlite.

what i see as a main benefit is that it is much harder to make mistakes writing custom keys, when i joined almost all mesh dbcode was subtly (and some not so much subtly) broken

Reply:
but what is the cost of adding them? why not add them? isn't it nice to be able to reap more benefit from sqlite?

Reply:
added for layers, but can't add them for signature and pubkey because genesis ballot doesn't have them and it messes up some tests.